### PR TITLE
Improve hxd.fmt.pak.loader

### PR DESCRIPTION
**Pull request by qkdreyer**
- Fixes https://github.com/qkdreyer/heaps-pak/commit/70b51614c870bf68f7e6aeda46195fc8febeab7b
- Handle progress bar loading for multiple pak files for sys system

### CODE REVIEW

**Comment by ncannasse** [comment]
Please rebase.
Also, getMaxCount will not work because BinaryLoader is async...

**Comment by qkdreyer** [comment]
I rebased it and reverted the BinaryLoader changes

**Comment by qkdreyer** [comment]
Please have a look again at this one

**Comment by ncannasse** [comment]
This will do a divide by 0 on non-sys targets (JS)
It's hard to tell in advance how many resources there is since JS is async. Also, loading is almost immediate in sys so i'm not sure what you're trying to achieve here.

File: hxd/fmt/pak/Loader.hx [hunk 1]
```diff
@@ -40,7 +42,7 @@ class Loader extends h2d.Object {
 
 			bg.x = (s2d.width - 100) >> 1;
 			bg.y = (s2d.height - 10) >> 1;
-			updateBG(0);
+			updateBG(resCount / this.maxResCount);
```
Comment (qkdreyer) [inline hxd/fmt/pak/Loader.hx:45]:
```suggestion
			updateBG(this.maxResCount > 0 ? resCount / this.maxResCount : 0);

```

**Comment by qkdreyer** [comment]
The main point here was about fixing this error :

image

And also trying to improve the progress bar state while loading multiple pak files

**Comment by ncannasse** [comment]
This error is no relevant to pak loader, I prefer to close for now ;)

**Pull request closed without merging**